sled-agent/src/services.rs

```diff
@@ -275,6 +275,12 @@ pub enum Error {
     #[error("Error querying simnet devices")]
     Simnet(#[from] GetSimnetError),
 
+    #[error("Failed to access storage")]
+    Storage(#[from] sled_storage::error::Error),
+
+    #[error("Missing datasets: {datasets:?}")]
+    MissingDatasets { datasets: HashSet<DatasetName> },
```

Comment:
Since the contents of the set will be printed out (e.g. in logs), perhaps it should be a `BTreeSet` rather than a `HashSet`, for consistent ordering? If I see a logged error like "missing datasets: {a, b, c}", it seems like it could be desirable to be able to grep for other occurrences of the same missing datasets error, which is difficult when the iteration order isn't stable...also, if they're always alphabetical, it's easier to write `grep` invocations or other regexes where you match any missing datasets error that includes at least some prefix/suffix of datasets.

Probably not a _huge_deal, but I figured I'd mention it.

Reply:
Done!